event-svc/src/event/validator/event.rs

```diff
@@ -0,0 +1,187 @@
+use std::sync::Arc;
+
+use ceramic_core::{Cid, EventId};
+use ceramic_event::unvalidated;
+use ipld_core::ipld::Ipld;
+use recon::ReconItem;
+use tokio::try_join;
+
+use crate::{
+    event::{service::InvalidItem, validator::signed::SignedValidator, DeliverableRequirement},
+    store::{EventInsertable, SqlitePool},
+    Result,
+};
+
+use super::grouped::{GroupedEvents, SignedEvents, Time};
+
+#[derive(Debug)]
+pub struct ValidatedEvents {
+    /// These events are valid
+    pub valid: Vec<ValidatedEvent>,
+    /// We don't have enough information to validate these events yet.
+    /// e.g. The init event is required and the node has not yet discovered it.
+    pub pending: Vec<UnvalidatedEvent>,
+    /// Events that failed validation
+    pub invalid: Vec<InvalidItem>,
+}
+
+#[derive(Debug)]
+pub struct ValidatedEvent {
+    key: EventId,
+    cid: Cid,
+    event: Arc<unvalidated::Event<Ipld>>,
+}
+
+impl ValidatedEvent {
+    pub fn order_key(&self) -> &EventId {
+        &self.key
+    }
+}
+
+// TODO: should this impl from or have a private method?
+// it's not supposed to be public outside the crate and it currently allows us to
+// skip validation for the cases we're not doing it
+impl From<UnvalidatedEvent> for ValidatedEvent {
+    fn from(value: UnvalidatedEvent) -> Self {
+        Self {
+            key: value.key,
+            cid: value.cid,
+            event: value.event,
+        }
+    }
+}
+
+impl From<ValidatedEvent> for EventInsertable {
+    fn from(value: ValidatedEvent) -> Self {
+        EventInsertable::new(value.key, value.cid, value.event, false)
+            .expect("validated events must be insertable")
+    }
+}
+
+#[derive(Debug)]
+pub struct UnvalidatedEvent {
```

Comment:
events can't really track their event ID can they? I mean, I guess we can assign one based on the network and state of the system but it's not part of the event so it feels a bit weird. I agree this name is weird, was just trying to wrap things together in something for the crate to use rather than have `(eventId, event)` tuples like we had all over. but with the CID stuff being known now, maybe this can be removed. I'll see what I can do.